app/utils/user-validation.ts

```diff
@@ -4,6 +4,7 @@ export const usernameSchema = z
 	.string()
 	.min(3, { message: 'Username is too short' })
 	.max(20, { message: 'Username is too long' })
+	.regex(new RegExp("^[^@]*$"), "username can not contains @")
```

Comment:
Wouldn't this work?

```suggestion
	.regex(/@/, 'Username can not contain @')
```

Reply:
does not work, we can avoid using `new RegExp(regular expression)` but only `/@/` is telling zod that the username should have the symbol `@`